src/main/java/radiounit/EricssonLteRadioUnitReceiver.java

```diff
@@ -13,60 +15,193 @@
+	 */
+	private volatile List<Carrier> carriers;
+	
+	public EricssonLteRadioUnitReceiver()
+	{
+		this.carriers = new ArrayList<Carrier>();
+	}
 
 	public void setupEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] setupEricssonLte");
 	}
 
 	public void activateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] activateEricssonLte");
 	}
 
 	public void deactivateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] deactivateEricssonLte");
 	}
 
 	public void releaseEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] releaseEricssonLte");
 	}
 
-	public void setupCarrierEricssonLte(Carrier carrrier) {
-		// TODO Auto-generated method stub
-
+	public void setupCarrierEricssonLte(Carrier carrier) {
+		System.out.println("[EricssonLteRadioUnitReceiver] setupCarrierEricssonLte: " + carrier);
+		
+		// Validate carrier
+		FrequencyBand band = carrier.getCarrierFrequencies();
+		boolean isLte = false;
+		
+		for (LteFrequencyBand lteFreq : LteFrequencyBand.values())
+		{
+			if (band == lteFreq)
+			{
+				isLte = true;
+				break;
+			}
+		}
+
+		if (!isLte)
+		{
+			System.err.println("Cannot add WCDMA carrier to LTE radio!");
+			return;
+		}
+		
+		if (getCarrierById(carrier.getCarrierId()) != null)
+		{
+			System.err.println("Carrier with id " + carrier.getCarrierId() + " already exists!");
+			return;
+		}
+		
+		addCarrierInternal(carrier);
 	}
 
 	public void signalScalingEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] signalScalingEricssonLte");
 	}
 
 	public void modifyCarrierEricssonLte(Integer carrierId, FrequencyBand frequencyBand) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] modifyCarrierEricssonLte: " + carrierId + ", " + frequencyBand.getBand());
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
+		{
+			System.err.println("Carrier with id " + carrierId + " does not exist!");
+			return;
+		}
+		
+		Carrier existingCarrier = getCarrierById(carrierId);
+		
+		removeCarrierInternal(idx);
+		
+		existingCarrier.setFrequencyBand(frequencyBand);
+		
+		addCarrierInternal(existingCarrier);
 	}
 
 	public void removeCarrierEricssonLte(Integer carrierId) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] removeCarrierEricssonLte: " + carrierId);
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
+		{
+			System.err.println("Carrier with id " + carrierId + " does not exist!");
+			return;
+		}
+		
+		removeCarrierInternal(idx);
 	}
 
 	public void selfDiagnosticsEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] selfDiagnosticsEricssonLte");
```

Comment:
This should be fine right? I couldn't find anywhere in the exam guidelines that stated what Ista wanted this method to actually accomplish.

Reply:
I believe so? Ya, when I looked through the requirements, there wasn't anything specific for this one (and a couple others) so just using the print statement for now.